apps/server/src/modules/tool/common/service/validation/rules/parameter-array-validator.ts

```diff
@@ -0,0 +1,6 @@
+import { ValidationError } from '@shared/common';
+import { CustomParameter, CustomParameterEntry } from '../../../domain';
+
+export interface ParameterArrayValidator {
+	validate(entries: CustomParameterEntry[], declarations: CustomParameter[]): ValidationError[];
```

Comment:
Nice! :) Good idea to split the whole validate function up since it was grown and grown